pygmt/src/grdhisteq.py

```diff
@@ -0,0 +1,366 @@
+        graytone will result in most of the image being very dark gray, with
+        the mountain being almost white. :meth:`pygmt.grdhisteq.compute_bins`
+        can provide a list of data values that divide the data range into
+        divisions which have an equal area in the image [Default is 16 if
+        ``divisions`` is not set]. The :class:`pandas.DataFrame` or ASCII file
+        output can be used to make a colormap with :meth:`pygmt.makecpt` and an
+        image with :meth:`pygmt.Figure.grdimage` that has all levels of gray
+        occuring equally.
+
+        Full option list at :gmt-docs:`grdhisteq.html`
+
+
+        Parameters
+        ----------
+        grid : str or xarray.DataArray
+            The file name of the input grid or the grid loaded as a DataArray.
+        outfile : str or bool or None
+            The name of the output ASCII file to store the results of the
+            histogram equalization in.
+        output_type : str
+            Determine the format the xyz data will be returned in [Default is
+            ``pandas``]:
+
+                - ``numpy`` - :class:`numpy.ndarray`
+                - ``pandas``- :class:`pandas.DataFrame`
+                - ``file`` - ASCII file (requires ``outfile``)
+        divisions : int
+            Set the number of divisions of the data range.
+        quadratic: bool
+            Perform quadratic equalization [Default is linear].
+        region : str or list
+            *xmin/xmax/ymin/ymax*\ [**+r**][**+u**\ *unit*].
+            Specify the :doc:`region </tutorials/basics/regions>` of interest.
+        verbose : bool or str
+            Select verbosity level [Default is **w**], which modulates the
+            messages written to stderr. Choose among 7 levels of verbosity:
+
+            - **q** - Quiet, not even fatal error messages are produced
+            - **e** - Error messages only
+            - **w** - Warnings [Default]
+            - **t** - Timings (report runtimes for time-intensive algorithms);
+            - **i** - Informational messages (same as ``verbose=True``)
+            - **c** - Compatibility warnings
+            - **d** - Debugging message
+
+
+        Returns
+        -------
+        ret: pandas.DataFrame None
+            Return type depends on the ``outfile`` parameter:
+
+            - pandas.DataFrame if ``outfile`` is True or None
+            - None if ``outfile`` is a str (file output is stored in
+              ``outfile``)
+
+        Example
+        -------
+        >>> import pygmt  # doctest: +SKIP
+        >>> # Load a grid of @earth_relief_30m data, with an x-range of 10 to
+        >>> # 30, and a y-range of 15 to 25
+        >>> grid = pygmt.datasets.load_earth_relief(
+        ...     resolution="30m", region=[10, 30, 15, 25]
+        ... )  # doctest: +SKIP
+        >>> # Find the elevation intervals that divide the grid into 5
+        >>> # divisions of equal area
```

Comment:
It actually is referring to spatial area for this example, since it is geographic. From the docs "For geographical grids we do a weighted histogram equalization since the area of each node varies with latitude."

Reply:
I'll add this note to our docstring as well.

Reply:
Right, I got a little confused by the sentence as the divisions are 1D elevation divisions (e.g. 0, 100m, 200m, 250m, 275m, etc), but the way those divisions are made is by dividing the 2D grid into 5 equal areas. Maybe we can reword the sentence a little bit?

Reply:
I revised this in https://github.com/GenericMappingTools/pygmt/pull/1433/commits/32757d88e916e2a1087d69e956481f29727ac3ac. Let me know if it's still not clear (it's a bit difficult to phrase right).